static/app/views/settings/projectIssueGrouping/upgradeGrouping.tsx

```diff
@@ -67,38 +102,35 @@ function UpgradeGrouping({
     } catch {
       handleXhrErrorResponse(t('Unable to upgrade config'));
     }
-  };
+  }
 
-  if (!groupingConfigs) {
-    return null;
+  function openUpgradeGroupingConfirmModal() {
+    if (location.hash !== `#${upgradeGroupingId}` || noUpdates || !groupingConfigs) {
+      return;
+    }
+
+    openConfirmModal({
+      confirmText,
+      priority,
+      onConfirm: handleConfirmUpgrade,
+      message: modalMessage,
+    });
   }
 
-  function getModalMessage() {
-    return (
-      <Fragment>
-        <TextBlock>
-          <strong>{t('Upgrade Grouping Strategy')}</strong>
-        </TextBlock>
-        <TextBlock>
-          {t(
-            'You can upgrade the grouping strategy to the latest but this is an irreversible operation.'
-          )}
-        </TextBlock>
-        <TextBlock>
-          <strong>{t('New Behavior')}</strong>
-          <div dangerouslySetInnerHTML={{__html: marked(updateNotes)}} />
-        </TextBlock>
-        <TextBlock>
-          <Alert type={alertType}>{riskNote}</Alert>
-        </TextBlock>
-      </Fragment>
-    );
+  function handleOpenUpgradeGroupingConfirmModal() {
```

Comment:
I could have reused `openUpgradeGroupingConfirmModal` but I always like to use the prefix handle when using a function in `onClick`

Reply:
honestly I would just call this method `openConfirmModal` since you're already in the context of this component. but 🤷